src/api/openThesaurusAPI.ts

```diff
@@ -0,0 +1,51 @@
+import { log } from "console";
+import type { PartOfSpeech, Synonym, SynonymProvider } from "src/api/types";
+
+class Base {
```

Comment:
You don't really need the `Base` Class if you only have one "resulting" Class at the end. It was only used, because the FreeDictionaryAPI has both a Definition- and SynonymProvider.